metadata.md

```diff
@@ -0,0 +1,98 @@
+# Metadata Schema - Version 1.0
+
+In order to ensure interoperability between implementations of the Arrow in-memory format
+and associated storage formats such as Parquet and Feather, it is necessary to
+include additional top-level metadata.
+
+This information includes:
+
+-   spatial properties of each geometry column
+-   name of the primary geometry column
+
+Metadata is to be stored as a JSON-encoded UTF-8 string under the "geo" key within the
+top-level metadata object of the in-memory or file format.
+
+For clarity, the following shows the unencoded JSON structure:
+
+```python
+"geo": {
+        "columns": {...},  # REQUIRED: see below
+        "creator": {
+            "library": "<library name>",  # REQUIRED
+            "version": "<library version>",  # REQUIRED
+        },
+        "primary_column": "<primary geometry column name>",  # REQUIRED
+        "version": "1.0.0"  # REQUIRED
+    }
+```
+
+Each geometry column in the dataset must be included in the `columns` field above
+with the following content, keyed by the column name:
+
+```python
+"<column_name>": {
+    "bounds": [<xmin>, <ymin>, (zmin), <xmax>, <ymax>, (zmax)],  # OPTIONAL
+    "crs": "<WKT representation of CRS>",  # REQUIRED
+    "encoding": "WKB",  # REQUIRED
```

Comment:
Maybe not needed for the initial PR? But I would also add an optional "dimension" and "geometry_type"

Reply:
What does `dimension` refer to?  As in x,y vs x,y,z coordinates?  How would that be serialized in a generic way?

What should `geometry_type` be when geometries in the column are of mixed type?

Reply:
If the encoding is WKB, then `dimension` and `geometry_type` are encoded at the row level, right?

Reply:
That is correct.

Reply:
Yes, so it probably only becomes relevant if allowing other encodings

Reply:
It might be a appropriate for us to develop an ancillary metadata structure to capture the unique details of a given encoder, when we all other encoders.  I.e., geometry type and dimension become important where they're not already encoded into the individual serialized geometry objects.  Likewise, for encodings that rely on all geometries having the same type and dimensions, those can be captured in this ancillary metadata.

For our first example of encoding a geometry column from geopandas, which can have mixed types and dimensions, it doesn't feel right to try and represent types and dimensions at the column level.

Reply:
I like the idea of having a standard place for metadata, so that a reader doesn't have to search for `metadata.geopandas.geometry_type`.

What about allowing `geometry_type` to be `null`, or `"mixed"` or `"row-level"`. Maybe if the file metadata has a non-null value for `geometry_type`, then that implies every geometry has the same type. But if `null`, then geometries are specified at the row level.

Reply:
`null` feels ambiguous since it shadows `row-level`.  But either way, this feels a little problematic, because we're using the `geometry_type` key to store the source of the geometry types, instead of a concrete geometry type - e.g., `Polygon`.  Or we're storing concrete types with sources, which doesn't seem right; you'd either have to mask the sources out or look for a specific list of well-known geometry types to make sense of the values in this field.

Reply:
I'm not sure I entirely understand. I was suggesting to choose one of those for optional support. So `geometry_type` could be `Point` or `LineString` or `Polygon` or potentially `Multi...`. In the case where all data in the table is known to be the same type, one of those would be set, and in the case where all data is not known to be of the same type, you might have `null`.

So essentially `geometry_type` is _either_ a concrete geometry type, and if not that implies there's no one concrete geometry type for the table.

Reply:
Ah, OK - sorry I misinterpreted.  I thought you were suggesting it could also have values `"mixed"` or `"row-level"`.  I see now you were suggesting one of those or `null`.

I think my preference is to wait on this bit until we've defined a bit more around other encodings, where we would need this information.  With WKB it is redundant, and since at least in `geopandas`, we aren't surfacing this metadata to users, I'm not yet sure how we'd use it.  

Presumably, anything in the metadata should be there to help the implementation for decoding the data back into a data frame, and not parallel information that is already available elsewhere within the data.

Reply:
Yes, I'm happy to wait on this and discuss along with other encodings.

Reply:
I think a `geometry_type` key could be optional, or have a default value of `null` or `"Unknown"` which would mean that the type is encoded in each value (eg as is the case with WKB). 

But indeed, that can wait until we need it.

BTW, even with WKB, it is not necessarily redundant. There might be applications that can find it useful to know that a certain dataset is a Point dataset, without needing to read in and decode all data.

Reply:
>For our first example of encoding a geometry column from geopandas, which can have mixed types and dimensions, it doesn't feel right to try and represent types and dimensions at the column level.

Wouldn't that be a `GeometryCollection` ?

Reply:
Not necessarily, because if geometry types can be specified at the row/feature level, each geometry can be specified as a concrete type. To my understanding, a `GeometryCollection` is when several different geometries share the same feature properties.

Reply:
> Wouldn't that be a GeometryCollection ?

GeoPandas currently allows that you have a column with mixed geometry type (eg both points, lines, polygons, geometrycollection etc in a single column). I think `sf` also supports this?

Reply:
Right, so `GeometryCollection` is at the row level and a column that stores more than one type at least postgis calls it a column of type `Geometry`.